### ISSUE TITLE
Documentation:  potential updates for derive_var_obs_number() example

### ISSUE THREAD

**Issue by kaz462** [pharmaverse/admiral#1561]
### Please select a category the issue is focused on?

Function Documentation

### Let us know where something needs a refresh or put your idea here!

To illustrate the default order of derive_var_obs_number() is ascending, could we add **desc**(VSTPTNUM) in the order parameter of its example? when deriving ASEQ, it happens a lot to have both descending and ascending order of variables

Also, the current example output from derive_var_obs_number() contains all missing values for the order variable (VSTPTNUM), maybe we could just remove the filter? 

Current:
![image](https://user-images.githubusercontent.com/47867131/199864663-cda94ac1-4bd8-40b1-9d22-ecd242082f81.png)

Update:
![image](https://user-images.githubusercontent.com/47867131/199866907-22df516a-4d0a-4e17-a1ad-434d888ec155.png)

**Comment by bms63** [comment]
Thank you for the feedback @kaz462 and the proposed updated example.  Would like to make this update for us?  It would be much appreciated!! :) Let us know if you need any assistance with working on it if you choose to do so.

**Comment by kaz462** [comment]
Thank you for your response @bms63  I'm happy to update. 
I want to check another trivial issue we faced related to **derive_var_obs_number**:

Issue:
My colleague @bmei1 is trying to validate our internal ADaM (developed in SAS) by using admiral-generated ADaM. While ordering variables to derive ASEQ, he found NA is at the beginning of ascending order column in the SAS data. But with derive_var_obs_number, NA is always at the bottom regardless of ascending or descending order -
this is from dplyr::arrange(), but there is a workaround in the arrange function by using is.na(), or !is.na()/desc(is.na()), so the missing values are able to go bottom or top based on users' need. This workaround does not apply to **derive_var_obs_number**,  could this feature be adapted?

Reason:
I found in the 3rd layer of derive_var_obs_number (->assert_order_vars-> is_order_vars) - **admiraldev::is_order_vars()** is the one limited this feature as it only allows desc() function. Could we add more functions in is_order_vars? (e.g., is.na, desc(is.na())) 

Documentation: 
The current description of the order parameter of derive_var_obs_number has Permitted Values: functions of variables - maybe it's worth mentioning which functions can be used here? 
![image](https://user-images.githubusercontent.com/47867131/200876979-4d59fce8-359c-4531-8e34-9d3aff7ea436.png)
